crypto/zk/circuit/config.go

```diff
@@ -0,0 +1,115 @@
+package circuit
+
+import (
+	"encoding/hex"
+	"log"
+	"strings"
+)
+
+// DefaultCircuitConfigurationTag constant contains the tag value that points
+// to the default ZkSnark circuit configuration. It ensures that at least one
+// circuit configuration is available so the configuration refered by this tag
+// must be defined.
+const DefaultCircuitConfigurationTag = "dev"
+
+// ZkCircuitConfig defines the configuration of the files to be downloaded
+type ZkCircuitConfig struct {
+	// URI defines the URI from where to download the files
+	URI string `json:"uri"`
+	// CircuitPath defines the path from where the files are downloaded
+	CircuitPath string `json:"circuitPath"`
+	// Levels refers the number of levels that the merkle tree associated to the
+	// current circuit configuration artifacts has
+	Levels int `json:"levels"`
+	// LocalDir defines in which directory will be the files
+	// downloaded, under that directory it will follow the CircuitPath
+	// directories structure
+	LocalDir string `json:"localDir,omitempty"`
+	// ProvingKeyHash contains the expected hash for the file filenameZKey
+	ProvingKeyHash []byte `json:"zKeyHash"`
+	// FilenameProvingKey defines the name of the file of the circom ProvingKey
+	ProvingKeyFilename string `json:"zKeyFilename"` // proving_key.zkey
+	// VerificationKeyHash contains the expected hash for the file filenameVK
+	VerificationKeyHash []byte `json:"vKeyHash"`
+	// FilenameVerificationKey defines the name of the file of the circom
+	// VerificationKey
+	VerificationKeyFilename string `json:"vKeyFilename"` // verification_key.json
+	// WasmHash contains the expected hash for the file filenameWasm
+	WasmHash []byte `json:"wasmHash"`
+	// FilenameWasm defines the name of the file of the circuit wasm compiled
+	// version
+	WasmFilename string `json:"wasmFilename"` // circuit.wasm
+}
+
+// KeySize returns the maximun number of bytes of a leaf key according to the
+// number of levels of the current circuit (nBytes = nLevels / 8).
+func (config ZkCircuitConfig) KeySize() int {
+	return config.Levels / 8
+}
+
+// CircuitsConfiguration stores the relation between the different vochain nets
+// and the associated circuit configuration. Any circuit configuration must have
+// the remote and local location of the circuits artifacts and their metadata
+// such as artifacts hash or the number of parameters.
+var CircuitsConfigurations = map[string]ZkCircuitConfig{
+	"bizono": {
```

Comment:
I'd not use these namings for the circuit configurations. We might have 2 dev networks but use the same circuit. We might also have 2 production networks and use the same circuit config.

For now I'd just use `dev`. Once we do the ceremony and so on, we might have something like `prod2023`